### PR TITLE
Add gif.js package

### PR DESCRIPTION
**Pull request by cprecioso**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## gif.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/gif.js@latest/))
was missing the following properties:
1. NeuQuant
1. TypedNeuQuant
1. GIFEncoder
1. LZWEncoder
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d4a855affefa7fe3dc4d33f5c263844e56120a47
</p>

File: types/gif.js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// Type definitions for gif.js 0.2
+// Project: https://github.com/jnordberg/gif.js#readme
+// Definitions by: Carlos Precioso <https://github.com/DefinitelyTyped>
```
Comment (peterblazejewicz) [inline types/gif.js/index.d.ts:3]:
```suggestion
// Definitions by: Carlos Precioso <https://github.com/cprecioso>
```

File: types/gif.js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { EventEmitter } from 'events';
+
+declare namespace GIF {
```
Comment (peterblazejewicz) [inline types/gif.js/index.d.ts:8]:
```suggestion
export as namespace GIF;

declare namespace GIF {
```
so it looks this is an UMD module, with 'GIF' exported:
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html#optional-global-usage
(read from script atteched to the page as per README)

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@cprecioso thx for the contribution!

**Comment by orta** [comment]
Agree, looks good to me 👍🏻

**Pull request merged**